[Merged by Bors] - Draft: Update CI configuration for macOS-13-xlarge

dsmello opened the pull request:
## Motivation
Improce the Pipelines times, migrating from the single self-hosted mac arm instance to the new GH m1 runner
Closes #

## Changes
Change the Pipeline runner from [self-hosted, mac, arm64] to  macOS-13-xlarge

## Test Plan
Run the entire CI

fasmat committed:
Improvements at Github Action: improve release names and add checksum (#5235)

Closes issue #5230
Closes PR #5234

Adjust the release pipeline to have better artifact names (more explicit From Windows to go-spacemesh-v1.2.4-win-amd64.zip ...)
And add the sha256 of each artifact generated and publish in this release

manual testing

- [ ] Update [changelog](../CHANGELOG.md) as needed

Co-authored-by: Davi Mello <dsmello.9@gmail.com>
Co-authored-by: unknown <andres@spacemesh.io>

dsmello committed:
Update CI configuration for macOS-13-xlarge

dsmello committed:
Update CI configuration for macOS-13-xlarge

dependabot[bot] committed:
build(deps): Bump github.com/spf13/viper from 1.17.0 to 1.18.0 (#5331)

Bumps [github.com/spf13/viper](https://github.com/spf13/viper) from 1.17.0 to 1.18.0.

pigmej committed:
Adjust from 10k default to 50k as default for gossip (#5323)

Better defaults for current network state.

dependabot[bot] committed:
build(deps): Bump actions/setup-go from 4 to 5 (#5332)

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 4 to 5.

dependabot[bot] committed:
build(deps): Bump github.com/spf13/viper from 1.18.0 to 1.18.1 (#5337)

Bumps [github.com/spf13/viper](https://github.com/spf13/viper) from 1.18.0 to 1.18.1.

fasmat committed:
Update CHANGELOG.md (#5333)

## Motivation
Several changes listed as part of the 1.3.0 release have already been released in prior versions.

## Changes
- Updated CHANGELOG to list changes at the correct versions
- renamed 1.3.0 back to UNRELEASED, since more changes might be added before it is officially released

## Test Plan
- n/a

## TODO
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

dependabot[bot] committed:
build(deps): Bump google-github-actions/setup-gcloud from 1 to 2 (#5336)

Bumps [google-github-actions/setup-gcloud](https://github.com/google-github-actions/setup-gcloud) from 1 to 2.

pigmej committed:
Increase default timeouts (#5338)

We're seeing a lot of `I/O deadline` and general timeouts currently in the network. Increasing the timeouts to 25s from standard 10s could help with some of them.

fasmat committed:
Prepare network for more nodes (#5339)

## Motivation
This is in anticipation for upcoming epochs, so that nodes can handle 300k+ active identities.

## Changes
- Increase sizes for various wire types
- Increase ATXHeader Cache to be able to hold the ATXs for Epoch 11 and Epoch 12 at the same time

## Test Plan
n/a

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

fasmat committed:
Revert part of #5339 (#5340)

## Motivation
Reverts part of #5339

## Changes
`PutUvarint` from standard library and `github.com/multiformats/go-varint` do not behave exactly the same. With Go 1.20 to the stdlib implementation was fixed to detect short reads, but in contrast to `github.com/multiformats/go-varint` it will still not return an error when decoding non-minimal encodings of integers.

## Test Plan
n/a

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

dependabot[bot] committed:
build(deps): Bump google.golang.org/grpc from 1.59.0 to 1.60.0 (#5342)

Bumps google.golang.org/grpc from 1.59.0 to 1.60.0.

dsmello committed:
Update macOS version in release workflow

dsmello committed:
Update macOS version in release workflow

fasmat committed:
Update certificates for tests using TLS (#5347)

## Motivation
The mTLS certificates used in testing of the `grpcserver` package expired recently. This PR includes new certificates that are valid for another 60 days.

This is a quick fix to unblock the failing CI, a proper fix (generating new certificates on every CI run) will follow.

## Changes
- generated new certificates using the instructions in README.md

## Test Plan
existing tests pass

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

dependabot[bot] committed:
build(deps): Bump k8s.io/client-go from 0.28.4 to 0.29.0 (#5351)

Bumps k8s.io/client-go from 0.28.4 to 0.29.0.

dependabot[bot] committed:
build(deps): Bump cloud.google.com/go/storage from 1.35.1 to 1.36.0 (#5350)

Bumps cloud.google.com/go/storage from 1.35.1 to 1.36.0.

dependabot[bot] committed:
build(deps): Bump github.com/google/uuid from 1.4.0 to 1.5.0 (#5345)

Bumps github.com/google/uuid from 1.4.0 to 1.5.0.

fasmat commented on .github/workflows/ci.yml:
```diff
@@ -143,7 +143,7 @@ jobs:
           - ubuntu-latest
           - [self-hosted, linux, arm64]
           - macos-latest
```
```suggestion
          - macos-12-large
```

pigmej replied:
Shouldn't it be `macos-13-large`, actually?

dsmello replied:
yes, I fixed

fasmat replied:
No, we want 12 actually. See my other comment.

dsmello replied:
I'll rollback to 12 because the issue mentioned before

fasmat replied:
We should use the same version for both our CI and our release build. Now we are using `macos-latest` for our CI but `macos-13-large` for our release?

fasmat commented on .github/workflows/ci.yml:
```diff
@@ -108,7 +108,7 @@ jobs:
           - ubuntu-latest
           - [self-hosted, linux, arm64]
           - macos-latest
```
```suggestion
          - macos-12-large
```

pigmej replied:
Shouldn't it be `macos-13-large`, actually?

fasmat replied:
No, because then we stop supporting macOS 12. `macos-latest` == `macos-12` we just want the bigger instance not drop support for macOS 12 🙂

fasmat committed:
Update dependencies not covered by dependabot (#5355)

## Motivation
Dependabot doesn't update dependencies that are not versioned. This PR updates those dependencies

## Changes
- update unversioned dependencies

## Test Plan
n/a

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

lrettig committed:
Fix typo/misspelling (#5357)

## Motivation
Fix a potentially confusing typo in a few places

## Changes
Only fixes typo, no logic changes

dependabot[bot] committed:
build(deps): Bump actions/download-artifact from 3 to 4 (#5358)

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 3 to 4.

poszu committed:
Increase p2p response data size to 40MiB (#5361)

## Motivation
Increase p2p response data size to 40MiB.

Also added log to show the actual data size in case of encoding failure.

dependabot[bot] committed:
build(deps): Bump google-github-actions/upload-cloud-storage from 1 to 2 (#5359)

Bumps [google-github-actions/upload-cloud-storage](https://github.com/google-github-actions/upload-cloud-storage) from 1 to 2.

dependabot[bot] committed:
build(deps): Bump actions/upload-artifact from 3 to 4 (#5360)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 3 to 4.

fasmat committed:
Generate TLS certificates as part of the test (#5356)

## Motivation
Follow up to #5347: instead of re-using generated templates create them as part of the test.

## Changes
- Update test to generate the certificates it needs

## Test Plan
- existing tests pass

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

poszu committed:
Autoscale post verifying workers (#5354)

## Motivation
Verifying POST proofs is CPU-intensive. By default, it uses 3/4 CPU cores. It is a problem during the cycle gap for the nodes that take longer to generate the POST proof than the others - new ATXs start to drop in and verifying them fights for CPU with the proving process. In such a situation, proving is more important so sacrificing ATX verification speed is an acceptable tradeoff. 

## Changes
Added a way to scale the number of workers that verify POST proofs (in incoming ATXs) up and down.

There is an auto scaler that reacts to POST events (started/finished) by scaling the number of workers down/up respectively.

The minimum and maximum number of workers is configurable with `smeshing-opts-verifying-min-workers` and `smeshing-opts-verifying-workers` respectively.

## Test Plan
Added unit tests.

## TODO
- [ ] Update [changelog](../CHANGELOG.md) as needed

dependabot[bot] committed:
build(deps): Bump github.com/spf13/viper from 1.18.1 to 1.18.2 (#5372)

Bumps [github.com/spf13/viper](https://github.com/spf13/viper) from 1.18.1 to 1.18.2.

dependabot[bot] committed:
build(deps): Bump golang.org/x/crypto from 0.16.0 to 0.17.0 (#5369)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.16.0 to 0.17.0.

ivan4th committed:
P2P decentralization improvements (#5329)

## Motivation

This PR implements changes needed for spacemeshos/pm#275, except for measurement

## Changes
* Introduce Routing Discovery to contact peers behind NATs
* Introduce dynamic v2 relay discovery which is needed for hole punching. The idea is to have a wider array of circuit-v2 passive relays which should be much safer than old libp2p active relays (which were disabled in e.g. Filecoin due to security concerns)
* Introduce QUIC transport to improve chances at hole punching, with testnet-mainnet "crosstalk" protection based on a transport-level handshake mechanism
  * the handshake is not used on mainnet. That way, connections between mainnet and testnet nodes are still prevented, as testnet peers expect the handshake, but if/when my libp2p changes are merged (libp2p/go-libp2p#2658) or libp2p gets private network support
* Make it possible to listen on multiple addresses and advertise multiple addresses
* Extend DebugService with additional P2P info needed for hole punching diagnostics (needs spacemeshos/api#285)
* Add `ping-peers` config option to facilitate P2P network issue diagnostics
* Add `force-dht-server` config option that is useful during troubleshooting DHT and hole-punching issues

`ping-peers` and `force-dht-server` were initially considered to be temporary features, but I think it might make sense to keep them for various P2P network troubleshooting scenarios.

All of the changes are disabled in the config by default, except for:
* libp2p Ping service is enabled by default to make diagnostics easier
* DHT Values and Providers as these will make DHT Routing Peer discovery work efficiently from the beginning when we enable this feature in the configs
* Bootnodes aren't used as relays by default anymore. v2 relays have very limited capacity by default and bootnode relay servers' reservations are very quickly exhausted. Need to either specify a static relay list or enable routing discovery, which searches for more available relays as needed

## Test Plan
* Tested using k8s several clusters with cone NATs enabled via `bridge` CNI plugin (via Multus) -- backported to v1.2.8
* Added a Mac node for testing

## TODO
- [x] Have spacemeshos/api#285 merged and updated to the new `api` release
- [x] Retest using an image based on this branch (not backport)
- [ ] Decide on whether/how to extend systests to include NAT testing
- [ ] To check: TCP holepunching tends to happen more than QUIC (might be related to the handshake mechanism)
- [ ] ~~To consider: try picking up some % (e.g.: 50%) of non-infra peers during routing discovery~~ (doesn't work too well, need something more involved for that)

Maybe as a follow-up (depending on how soon this gets reviewed):
- Include new metrics / check if they're already present
  - NAT type (UDP / TCP) - Cone / Symmetric / Unknown
  - Reachability - Public / Private / Unknown
  - N of "advertised" peers found via routing discovery
  - N of TCP and UDP (QUIC) peers
  - N of peers reached via relayed connections (these being present for a long time may indicate hole-punching troubles, usually relayed connections go away relatively quickly)
  - N of relay reservations this node managed to obtain
  - Whether routing discovery is active or suspended (e.g. b/c `low-peers` N of peers has been reached)
  - Whether DHT is in the `Server` or `Client` mode
- systests checking NATed connections

Co-authored-by: Ivan Shvedunov <ivan4th@users.noreply.github.com>

dependabot[bot] committed:
build(deps): Bump google.golang.org/grpc from 1.60.0 to 1.60.1 (#5371)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.60.0 to 1.60.1.

dependabot[bot] committed:
build(deps): Bump github.com/quic-go/quic-go from 0.39.4 to 0.40.1 (#5381)

Bumps [github.com/quic-go/quic-go](https://github.com/quic-go/quic-go) from 0.39.4 to 0.40.1.

poszu committed:
Avoid processing same ATX in parallel (#5379)

## Motivation
There is no mechanism preventing duplicated processing of ATX in parallel. An ATX might be gossiped multiple times while the same one is already being processed. Related issue: https://github.com/spacemeshos/go-spacemesh/issues/4426

## Changes
The ATX handler now registers the ATXID when it starts processing it. The subsequent calls to `handleATX()` will find that the handler is already processing this ATX and will register themself for the result, which is sent over a channel after the ATX processing is finished.

## Test Plan
Added a UT in which an ATX is gossiped 10x in parallel. The expectation is that it's processed only once.

ivan4th committed:
Fix handling "fetcher" log level (#5368)

"fetcher" key was being ignored in the "logging" section

ivan4th committed:
Add no-main-override config option (#5367)

This makes it possible to force "nomain" builds to run on mainnet.

## Motivation

Sometimes it is necessary to try a "nomain" build with mainnet.

## Changes

This adds `no-main-override` toplevel config option (bool) and also `--no-main-override` command line flag.

ivan4th committed:
Make disable-connection-manager option really work (#5370)

Need to provide `NullConnMgr` as the connection manager for no peer trimming to happen. This is useful for debugging problems that may be related to peer trimming by the libp2p `ConnectionManager`.

dependabot[bot] committed:
build(deps): Bump go.uber.org/mock from 0.3.0 to 0.4.0 (#5386)

Bumps [go.uber.org/mock](https://github.com/uber/mock) from 0.3.0 to 0.4.0.

dshulyak committed:
spawn a process to download atxs hinted by a server (#5377)

related: https://github.com/spacemeshos/go-spacemesh/issues/5366

this change disables sync reconciliation at the end and start of epoch, normal atx downloading works the way it was working.
instead it will ask peers for atxs that are recorded in the set, shared by the server.

fasmat committed:
Extend proposal builder to allow the use of a fallback active set. (#5378)

## Motivation
related to #5366 

Allow to set a fallback active set to be used by the proposal builder via the same mechanism as we use for the fallback beacon.

## Changes
- some minor fixes in `bootstrap` cmd code
- extend `ProposalBuilder` to be able to set a specific active set for an epoch
- extend node routine listening for updates to fallback beacon / activeset to propagate a fallback to the `ProposalBuilder`

## Test Plan
- added test case to `ProposalBuilder` to verify use of fallback active set if available
- manual testing on `testnet-10`

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

dependabot[bot] committed:
build(deps): Bump google.golang.org/protobuf from 1.31.0 to 1.32.0 (#5392)

Bumps google.golang.org/protobuf from 1.31.0 to 1.32.0.

dependabot[bot] committed:
build(deps): Bump github.com/prometheus/client_golang from 1.17.0 to 1.18.0 (#5395)

Bumps https://github.com/prometheus/client_golang from 1.17.0 to 1.18.0.

dsmello committed:
Merge branch 'develop' into gh-action/test-new-runners

fasmat commented on .github/workflows/ci.yml:
```diff
@@ -182,6 +185,9 @@ jobs:
       - name: Add OpenCL support - Windows
         if: ${{ matrix.os == 'windows-latest' }}
         run: choco install opencl-intel-cpu-runtime
+      - name: Install coreutils
+        if: ${{ matrix.os == 'macos-latest' }} || ${{ matrix.os == 'macos-13-xlarge' }}
+        run: brew install coreutils
```
This shouldn't be needed.

fasmat commented on .github/workflows/release.yml:
```diff
@@ -20,9 +20,10 @@ jobs:
           - os: [self-hosted, linux, arm64]
             outname_sufix: "linux-arm64"
-          - os: macos-latest
+          - os: macos-13-large
```
See my comment in `ci.yml`; do we want to drop support for macOS 12 already?

dsmello replied:
I'm rolling back to mac12

dsmello committed:
Update CI configuration for macOS builds

dsmello committed:
Update macOS instance size in CI workflow

fasmat commented on .github/workflows/ci.yml:
```diff
@@ -107,13 +107,18 @@ jobs:
         if: ${{ matrix.os == 'ubuntu-latest' }}
         run: sudo apt-get update -q && sudo apt-get install -qy ocl-icd-opencl-dev libpocl2
+
+      - name: Install coreutils
+        if: ${{ matrix.os == 'macos-12-large' || matrix.os == 'macos-13-xlarge' }}
+        run: brew install coreutils
```
Why this additional step in the CI workflow?

dsmello replied:
I didnt know if we would needed on the new runner. I'll remove. 1s

dsmello replied:
```suggestion
```

dsmello committed:
Update .github/workflows/ci.yml

fasmat commented:
bors merge
